File: 2021/src/day01.rs
use crate::utils;

pub fn day01() {
    let numbers =
        utils::convert_string_to_int_vector(utils::read_input_file("src/day01_input.txt").unwrap());
    let mut result = 0;
    for index in 1..numbers.len() {
        if numbers[index] > numbers[index - 1] {
            result = result + 1;
        }
    }

    println!("{}", result);
}

pub fn day01_part2() {
    let numbers =
        utils::convert_string_to_int_vector(utils::read_input_file("src/day01_input.txt").unwrap());

    let mut result = 0;

    for index in 3..numbers.len() {
        if numbers[index] > numbers[index - 3] {
            result = result + 1;
        }
    }

    println!("{}", result);
}

File: 2021/dart/lib/day01.dart
import 'utils.dart' as utils;

int core(int initialIndex) {
  var lines = utils.readFile("../inputs/day01.txt");
  var increased = 0;

  for (int index = initialIndex; index < lines.length; index++) {
    int currentNumber = int.parse(lines[index]);
    int prevNumber = int.parse(lines[index - initialIndex]);

    if (currentNumber > prevNumber) {
      increased++;
    }
  }

  return increased;
}

int day01() {
  return core(1);
}

int day01Part2() {
  return core(3);
}
